### PR TITLE
Functions for elliptic curves

### PR DESCRIPTION
**Pull request by delphinepol**
Added functions for elliptic curves based on the corresponding functions in Hecke

### CODE REVIEW

File: experimental/PlaneCurve/ProjEllipticCurve.jl [hunk 1]
```diff
@@ -457,3 +445,94 @@ function toweierstrass(C::ProjPlaneCurve{S}, P::Oscar.Geometry.ProjSpcElem{S}) w
...
+end
+
+################################################################################
+#
+# Interface calling the functions in Hecke
+#
+################################################################################
+
+@doc Markdown.doc"""
+    order(E::ProjEllipticCurve)
```
Comment (fingolfin) [inline experimental/PlaneCurve/ProjEllipticCurve.jl:463]:
```suggestion
    order(E::ProjEllipticCurve)

```

File: experimental/PlaneCurve/ProjEllipticCurve.jl [hunk 2]
```diff
@@ -457,3 +445,94 @@ function toweierstrass(C::ProjPlaneCurve{S}, P::Oscar.Geometry.ProjSpcElem{S}) w
...
+@doc Markdown.doc"""
+    rand(E::ProjEllipticCurve)
+
+Return a random point on the elliptic curve `E` defined over a finite field.
+"""
+function Oscar.rand(E::ProjEllipticCurve)
+   PP = proj_space(E)
+   H = E.Hecke_ec
+   if !Hecke.isshort(H)
+      L = Hecke.short_weierstrass_model(H)
```
Comment (fingolfin) [inline experimental/PlaneCurve/ProjEllipticCurve.jl:483]: Isn't it rather expensive to call `short_weierstrass_model` again and again if one wants to e.g. sample 100 different points on the curve?

**Review by fingolfin — APPROVED** [review]
Seems reasonable to me, though perhaps the people who wrote the code in Hecke (@thofma I guess?) may want to comment... or maybe not, dunno :-)

File: experimental/PlaneCurve/ProjEllipticCurve.jl (hunk 2 above)
Comment (delphinepol) [reply experimental/PlaneCurve/ProjEllipticCurve.jl:483]: Good remark, I added a function which enables to compute a list of random points.

**Comment by fingolfin** [comment]
Closing this as it is part of #470

**Pull request closed without merging**